src/OpenTelemetry.Extensions.Enrichment/TraceEnrichmentBag.cs

```diff
@@ -0,0 +1,36 @@
+// <copyright file="TraceEnrichmentBag.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Diagnostics;
+
+namespace OpenTelemetry.Extensions.Enrichment;
+
+#pragma warning disable CA1815 // Override equals and operator equals on value types
+public struct TraceEnrichmentBag
+#pragma warning restore CA1815 // Override equals and operator equals on value types
+{
+    public TraceEnrichmentBag(Activity activity)
+    {
+        this.Activity = activity;
+    }
+
+    public Activity Activity { get; init; }
```

Comment:
the usage is not clear to me from this.. The TraceEnrichmentBag allows users to add KVP which will be used to enrich the underlying trace/activity.. So why would we need to expose the activity publicly? if user has Activity access, they can directly do the enrichment, no?

Reply:
Yes, you are right. However, this is done for convenience to allow users to enrich traces depending on information they see in Activity. cc @dpk83

Reply:
It feels confusing. If a user has activity, then I'd simply suggest to call `activity.SetTag`.

> enrich traces depending on information they see in Activity
I suggest to not expose this until there is a solid use case. The OTel spec defines Span API as write only (which Activity does not follow as it allow read/writes), so the ability to "read something from span, and then use that to further add more tags to span" should not be a requirement.

Things like checking `IsRecording` before adding attributes - I think this should be covered by this enrichment framework. If we expect end users to do that, then this is basically providing 2 ways to achieve the same (activity.SetTag, enrichmentbag.SetTag).

Reply:
That's because we want to provide the users with the ability to use the existing information in Activity to make decision to enrich a certain piece of information or not in their enrichers. e.g we will also expose the Instrument from the metric enricher.

Also, the users that use this enrichment model will only use this enrichment model so I don't see a reason for confusion. Again if the users directly wants to call SetTag on the activity they are free to do so, but the important thing is that the solution is complete and user can make their enrichment decisions based on the additional context.

Reply:
_That's because we want to provide the users with the ability to use the existing information in Activity to make decision to enrich a certain piece of information or not in their enrichers_

I suggest to start without it, and add it, if there is a concrete need.

Reply:
> I suggest to start without it, and add it, if there is a concrete need.

That's a fair ask, @evgenyfedorov2 let's do that

Reply:
Made it private